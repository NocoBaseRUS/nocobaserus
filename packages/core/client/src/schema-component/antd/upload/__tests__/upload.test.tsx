import React from 'react';
import { render } from 'testUtils';
import App1 from '../demos/demo1';
import App2 from '../demos/demo2';

describe('Upload', () => {
  it('basic', () => {
    const { container } = render(<App1 />);
    expect(container).toMatchSnapshot();
  });

  it('uploading', () => {
    const { container } = render(<App2 />);
    expect(container).toMatchSnapshot();
  });
});
