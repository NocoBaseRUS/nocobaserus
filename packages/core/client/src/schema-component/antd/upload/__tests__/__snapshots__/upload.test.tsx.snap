// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Upload > basic 1`] = `
<div>
  <div
    class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-loose ant-formily-item-label-align-right ant-formily-item-control-align-left"
  >
    <div
      class="ant-formily-item-label"
    >
      <div
        class="ant-formily-item-label-content"
      >
        <span>
          <label>
            编辑模式
          </label>
        </span>
      </div>
      <span
        class="ant-formily-item-colon"
      >
        :
      </span>
    </div>
    <div
      class="ant-formily-item-control"
    >
      <div
        class="ant-formily-item-control-content"
      >
        <div
          class="ant-formily-item-control-content-component"
        >
          <div>
            <div
              class="ant-upload-picture-card-wrapper nb-upload"
            >
              <div
                class="ant-upload-list ant-upload-list-picture-card"
              >
                <div
                  class="ant-upload-list-picture-card-container"
                >
                  <span
                    class="ant-upload-picture-card-wrapper"
                  >
                    <div
                      class="ant-upload ant-upload-select ant-upload-select-picture-card"
                    >
                      <span
                        class="ant-upload"
                        role="button"
                        tabindex="0"
                      >
                        <input
                          accept=""
                          style="display: none;"
                          type="file"
                        />
                        <span>
                          <span
                            aria-label="plus"
                            class="anticon anticon-plus"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="plus"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <defs>
                                <style />
                              </defs>
                              <path
                                d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                              />
                              <path
                                d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                              />
                            </svg>
                          </span>
                          <br />
                           
                          Upload
                        </span>
                      </span>
                    </div>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-loose ant-formily-item-label-align-right ant-formily-item-control-align-left"
  >
    <div
      class="ant-formily-item-label"
    >
      <div
        class="ant-formily-item-label-content"
      >
        <span>
          <label>
            阅读模式
          </label>
        </span>
      </div>
      <span
        class="ant-formily-item-colon"
      >
        :
      </span>
    </div>
    <div
      class="ant-formily-item-control"
    >
      <div
        class="ant-formily-item-control-content"
      >
        <div
          class="ant-formily-item-control-content-component"
        >
          <div>
            <div
              class="ant-upload-picture-card-wrapper nb-upload"
            >
              <div
                class="ant-upload-list ant-upload-list-picture-card"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Upload > uploading 1`] = `
<div>
  <div
    class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-loose ant-formily-item-label-align-right ant-formily-item-control-align-left"
  >
    <div
      class="ant-formily-item-label"
    >
      <div
        class="ant-formily-item-label-content"
      >
        <span>
          <label>
            编辑模式
          </label>
        </span>
      </div>
      <span
        class="ant-formily-item-colon"
      >
        :
      </span>
    </div>
    <div
      class="ant-formily-item-control"
    >
      <div
        class="ant-formily-item-control-content"
      >
        <div
          class="ant-formily-item-control-content-component"
        >
          <div>
            <div
              class="ant-upload-picture-card-wrapper nb-upload"
            >
              <div
                class="ant-upload-list ant-upload-list-picture-card"
              >
                <div
                  class="ant-upload-list-picture-card-container"
                >
                  <div
                    class="ant-upload-list-item ant-upload-list-item-done ant-upload-list-item-list-type-picture-card"
                  >
                    <div
                      class="ant-upload-list-item-info"
                    >
                      <span
                        class="ant-upload-span"
                      >
                        <a
                          class="ant-upload-list-item-thumbnail"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/cd48dc833ab01aa3959ac39309fc39de.jpg"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <img
                            alt="s33766399"
                            class="ant-upload-list-item-image"
                            src="https://nocobase.oss-cn-beijing.aliyuncs.com/cd48dc833ab01aa3959ac39309fc39de.jpg"
                          />
                        </a>
                        <a
                          class="ant-upload-list-item-name"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/cd48dc833ab01aa3959ac39309fc39de.jpg"
                          rel="noopener noreferrer"
                          target="_blank"
                          title="s33766399"
                        >
                          Uploading
                        </a>
                      </span>
                    </div>
                    <span
                      class="ant-upload-list-item-actions"
                    >
                      <div
                        class="ant-space ant-space-horizontal ant-space-align-center"
                      >
                        <div
                          class="ant-space-item"
                          style="margin-right: 3px;"
                        >
                          <button
                            class="ant-btn ant-btn-text ant-btn-sm ant-btn-icon-only"
                            type="button"
                          >
                            <span
                              aria-label="download"
                              class="anticon anticon-download"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="download"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M505.7 661a8 8 0 0012.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9h-74.1V168c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v338.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.8zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                        <div
                          class="ant-space-item"
                        >
                          <button
                            class="ant-btn ant-btn-text ant-btn-sm ant-btn-icon-only"
                            type="button"
                          >
                            <span
                              aria-label="delete"
                              class="anticon anticon-delete"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="delete"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                      </div>
                    </span>
                    <div
                      class="ant-upload-list-item-progress"
                    >
                      <div
                        class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default"
                        role="progressbar"
                      >
                        <div
                          class="ant-progress-outer"
                        >
                          <div
                            class="ant-progress-inner"
                          >
                            <div
                              class="ant-progress-bg"
                              style="width: 60%; height: 2px;"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-upload-list-picture-card-container"
                >
                  <div
                    class="ant-upload-list-item ant-upload-list-item-done ant-upload-list-item-list-type-picture-card"
                  >
                    <div
                      class="ant-upload-list-item-info"
                    >
                      <span
                        class="ant-upload-span"
                      >
                        <a
                          class="ant-upload-list-item-thumbnail"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/d9f6ad6669902a9a8a1229d9f362235a.docx"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <img
                            alt="简历"
                            class="ant-upload-list-item-image"
                            src="//img.alicdn.com/tfs/TB1n8jfr1uSBuNjy1XcXXcYjFXa-200-200.png"
                          />
                        </a>
                        <a
                          class="ant-upload-list-item-name"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/d9f6ad6669902a9a8a1229d9f362235a.docx"
                          rel="noopener noreferrer"
                          target="_blank"
                          title="简历"
                        >
                          简历
                        </a>
                      </span>
                    </div>
                    <span
                      class="ant-upload-list-item-actions"
                    >
                      <div
                        class="ant-space ant-space-horizontal ant-space-align-center"
                      >
                        <div
                          class="ant-space-item"
                          style="margin-right: 3px;"
                        >
                          <button
                            class="ant-btn ant-btn-text ant-btn-sm ant-btn-icon-only"
                            type="button"
                          >
                            <span
                              aria-label="download"
                              class="anticon anticon-download"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="download"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M505.7 661a8 8 0 0012.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9h-74.1V168c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v338.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.8zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                        <div
                          class="ant-space-item"
                        >
                          <button
                            class="ant-btn ant-btn-text ant-btn-sm ant-btn-icon-only"
                            type="button"
                          >
                            <span
                              aria-label="delete"
                              class="anticon anticon-delete"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="delete"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                      </div>
                    </span>
                  </div>
                </div>
                <div
                  class="ant-upload-list-picture-card-container"
                >
                  <span
                    class="ant-upload-picture-card-wrapper"
                  >
                    <div
                      class="ant-upload ant-upload-select ant-upload-select-picture-card"
                    >
                      <span
                        class="ant-upload"
                        role="button"
                        tabindex="0"
                      >
                        <input
                          accept=""
                          multiple=""
                          style="display: none;"
                          type="file"
                        />
                        <span>
                          <span
                            aria-label="plus"
                            class="anticon anticon-plus"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="plus"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <defs>
                                <style />
                              </defs>
                              <path
                                d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                              />
                              <path
                                d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                              />
                            </svg>
                          </span>
                          <br />
                           
                          Upload
                        </span>
                      </span>
                    </div>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-loose ant-formily-item-label-align-right ant-formily-item-control-align-left"
  >
    <div
      class="ant-formily-item-label"
    >
      <div
        class="ant-formily-item-label-content"
      >
        <span>
          <label>
            阅读模式
          </label>
        </span>
      </div>
      <span
        class="ant-formily-item-colon"
      >
        :
      </span>
    </div>
    <div
      class="ant-formily-item-control"
    >
      <div
        class="ant-formily-item-control-content"
      >
        <div
          class="ant-formily-item-control-content-component"
        >
          <div>
            <div
              class="ant-upload-picture-card-wrapper nb-upload"
            >
              <div
                class="ant-upload-list ant-upload-list-picture-card"
              >
                <div
                  class="ant-upload-list-picture-card-container"
                >
                  <div
                    class="ant-upload-list-item ant-upload-list-item-done ant-upload-list-item-list-type-picture-card"
                  >
                    <div
                      class="ant-upload-list-item-info"
                    >
                      <span
                        class="ant-upload-span"
                      >
                        <a
                          class="ant-upload-list-item-thumbnail"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/cd48dc833ab01aa3959ac39309fc39de.jpg"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <img
                            alt="s33766399"
                            class="ant-upload-list-item-image"
                            src="https://nocobase.oss-cn-beijing.aliyuncs.com/cd48dc833ab01aa3959ac39309fc39de.jpg?x-oss-process=style/thumbnail"
                          />
                        </a>
                        <a
                          class="ant-upload-list-item-name"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/cd48dc833ab01aa3959ac39309fc39de.jpg"
                          rel="noopener noreferrer"
                          target="_blank"
                          title="s33766399"
                        >
                          s33766399
                        </a>
                      </span>
                    </div>
                    <span
                      class="ant-upload-list-item-actions"
                    >
                      <div
                        class="ant-space ant-space-horizontal ant-space-align-center"
                      >
                        <div
                          class="ant-space-item"
                        >
                          <button
                            class="ant-btn ant-btn-text ant-btn-sm ant-btn-icon-only"
                            type="button"
                          >
                            <span
                              aria-label="download"
                              class="anticon anticon-download"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="download"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M505.7 661a8 8 0 0012.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9h-74.1V168c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v338.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.8zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                      </div>
                    </span>
                  </div>
                </div>
                <div
                  class="ant-upload-list-picture-card-container"
                >
                  <div
                    class="ant-upload-list-item ant-upload-list-item-done ant-upload-list-item-list-type-picture-card"
                  >
                    <div
                      class="ant-upload-list-item-info"
                    >
                      <span
                        class="ant-upload-span"
                      >
                        <a
                          class="ant-upload-list-item-thumbnail"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/d9f6ad6669902a9a8a1229d9f362235a.docx"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <img
                            alt="简历"
                            class="ant-upload-list-item-image"
                            src="//img.alicdn.com/tfs/TB1n8jfr1uSBuNjy1XcXXcYjFXa-200-200.png?x-oss-process=style/thumbnail"
                          />
                        </a>
                        <a
                          class="ant-upload-list-item-name"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/d9f6ad6669902a9a8a1229d9f362235a.docx"
                          rel="noopener noreferrer"
                          target="_blank"
                          title="简历"
                        >
                          简历
                        </a>
                      </span>
                    </div>
                    <span
                      class="ant-upload-list-item-actions"
                    >
                      <div
                        class="ant-space ant-space-horizontal ant-space-align-center"
                      >
                        <div
                          class="ant-space-item"
                        >
                          <button
                            class="ant-btn ant-btn-text ant-btn-sm ant-btn-icon-only"
                            type="button"
                          >
                            <span
                              aria-label="download"
                              class="anticon anticon-download"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="download"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M505.7 661a8 8 0 0012.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9h-74.1V168c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v338.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.8zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-loose ant-formily-item-label-align-right ant-formily-item-control-align-left"
  >
    <div
      class="ant-formily-item-label"
    >
      <div
        class="ant-formily-item-label-content"
      >
        <span>
          <label>
            小图预览
          </label>
        </span>
      </div>
      <span
        class="ant-formily-item-colon"
      >
        :
      </span>
    </div>
    <div
      class="ant-formily-item-control"
    >
      <div
        class="ant-formily-item-control-content"
      >
        <div
          class="ant-formily-item-control-content-component"
        >
          <div>
            <div
              class="ant-upload-picture-card-wrapper nb-upload nb-upload-small"
            >
              <div
                class="ant-upload-list ant-upload-list-picture-card"
              >
                <div
                  class="ant-upload-list-picture-card-container"
                >
                  <div
                    class="ant-upload-list-item ant-upload-list-item-done ant-upload-list-item-list-type-picture-card"
                  >
                    <div
                      class="ant-upload-list-item-info"
                    >
                      <span
                        class="ant-upload-span"
                      >
                        <a
                          class="ant-upload-list-item-thumbnail"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/cd48dc833ab01aa3959ac39309fc39de.jpg"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <img
                            alt="s33766399"
                            class="ant-upload-list-item-image"
                            src="https://nocobase.oss-cn-beijing.aliyuncs.com/cd48dc833ab01aa3959ac39309fc39de.jpg?x-oss-process=style/thumbnail"
                          />
                        </a>
                        <a
                          class="ant-upload-list-item-name"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/cd48dc833ab01aa3959ac39309fc39de.jpg"
                          rel="noopener noreferrer"
                          target="_blank"
                          title="s33766399"
                        >
                          s33766399
                        </a>
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-upload-list-picture-card-container"
                >
                  <div
                    class="ant-upload-list-item ant-upload-list-item-done ant-upload-list-item-list-type-picture-card"
                  >
                    <div
                      class="ant-upload-list-item-info"
                    >
                      <span
                        class="ant-upload-span"
                      >
                        <a
                          class="ant-upload-list-item-thumbnail"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/d9f6ad6669902a9a8a1229d9f362235a.docx"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <img
                            alt="简历"
                            class="ant-upload-list-item-image"
                            src="//img.alicdn.com/tfs/TB1n8jfr1uSBuNjy1XcXXcYjFXa-200-200.png?x-oss-process=style/thumbnail"
                          />
                        </a>
                        <a
                          class="ant-upload-list-item-name"
                          href="https://nocobase.oss-cn-beijing.aliyuncs.com/d9f6ad6669902a9a8a1229d9f362235a.docx"
                          rel="noopener noreferrer"
                          target="_blank"
                          title="简历"
                        >
                          简历
                        </a>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
